stdlib/public/core/Collection.swift

```diff
@@ -1338,6 +1338,23 @@ extension Collection {
       offsetBy: numericCast(amount), limitedBy: endIndex) ?? endIndex
     return self[startIndex..<end]
   }
+  
+  /// Returns a subsequence by skipping elements while `predicate` returns
+  /// `true` and returning the remaining elements.
+  ///
+  /// - Parameter while: A closure that takes an element of the
+  ///   sequence as its argument and returns `true` if the element should
+  ///   be skipped or `false` if it should be included. Once the predicate
+  ///   returns `false` it will not be called again.
+  public func drop(
+    while predicate: @noescape (Iterator.Element) throws -> Bool
+  ) rethrows -> SubSequence {
+    var start = startIndex
+    while try start < endIndex && predicate(self[start]) {
```

Comment:
`start != endIndex` can be cheaper than `<`.  It is usually the same, but in some cases can be cheaper by a constant factor.